src/kernels/kernelFinder.ts

```diff
@@ -61,7 +61,11 @@ export class KernelFinder implements IKernelFinder {
         const kernels: KernelConnectionMetadata[] = [];
 
         for (const finder of this._finders) {
-            const contributedKernels = finder.listContributedKernels(resource);
+            const contributedKernels = finder.listContributedKernels(resource).map((kernelConnection) => {
```

Comment:
@DonJayamanne Most of this change is all experimental code at the UI level under controllers. However there were a couple small changes at the KernelFinder. I want to make sure that I have your ok with how I'm handling this. 

We can chat online if needed.

This spot here is where I tag the kernel connections with the kernelFinderInfo that contributed them. I added this kernelFinderInfo as an optional thing on the type.

If desired I could check the configuration here to see if the experiment is active to see if this should be done or not.